Update i18n guide to accompany docs PR 9896 (#31)
withastro/contribute.docs.astro.build: src/content/docs/guides/i18n.mdx

For example, the English language “Getting Started” page is at `src/content/docs/en/getting-started.mdx` and the same page in French is at `src/content/docs/fr/getting-started.mdx`.

#### Reusable UI strings

We also have translations for UI text strings. These are short bits of text used to label or structure components of the documentation UI. For example, the text “Sponsored by” before our sponsor logos on the English pages is translated so we can show “Sponsorisé par” on the French pages instead.

UI strings are stored in the `src/content/i18n/` folder, with a dedicated YAML file for each language. Unlike pages, these translations look more like a dictionary, mapping standard keys to translated strings. Comments can be included on a line starting with a `#`.

The following example shows an abbreviated version of the English UI strings:

```yml title="src/content/i18n/en.yml"
a11y.sectionLink: Section titled
# Site settings
site.title: Astro Documentation
# Left Sidebar
leftSidebar.sponsoredBy: Sponsored by
# ...
```

> Keys are not translated, only the English text. So the example above is translated into French is:

Author: @sarah11918 Aren’t the keys also in English? 😅

Reviewer: Maybe something like this to further clarify?

```suggestion
Keys, although they are written in English, should not be translated. The only part that should be translated is the text following the key. For example, when translated into French, it appears like this:
```

Author: I like that! But I do think it’s a bit much when just explaining a the example. We can probably simplify back to something like the original:
```suggestion
Keys are not translated, so the example above is translated into French as:
```

> Keys are not translated, only the English text. So the example above is translated into French is:

Reviewer: Maybe something like this to further clarify?

```suggestion
Keys, although they are written in English, should not be translated. The only part that should be translated is the text following the key. For example, when translated into French, it appears like this:
```
